x-pack/legacy/plugins/snapshot_restore/public/app/services/validation/validate_restore.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { RestoreSettings } from '../../../../common/types';
+import { UNMODIFIABLE_INDEX_SETTINGS, UNREMOVABLE_INDEX_SETTINGS } from '../../../app/constants';
+import { textService } from '../text';
+
+export interface RestoreValidation {
+  isValid: boolean;
+  errors: { [key: string]: React.ReactNode[] };
+}
+
+const isStringEmpty = (str: string | null): boolean => {
+  return str ? !Boolean(str.trim()) : true;
+};
+
+export const validateRestore = (restoreSettings: RestoreSettings): RestoreValidation => {
```

Comment:
I think this file is a great candidate for some unit tests.

Reply:
will add in a follow up